### ISSUE TITLE
Sonic and All-Stars Racing Transformed Collection: DxvkMemoryAllocator: Memory allocation failed

### ISSUE THREAD

**Issue by vinifmor** [Joshua-Ashton/d9vk#221]
The game crashes when loading the main window for **d9vk 0.13**. For **0.12** and **0.11** it crashes when a race starts (black screen).  The game runs well without d9vk set.

### System information
- GPU: GeForce 740M
- Driver: Nvidia 430.26
- Wine version: Proton 4.2-9
- DXVK version: d9vk 0.13

### Proton log (just the end of the file):
5873.915:002b:0034:trace:seh:call_vectored_handlers calling handler at 0x6f46fa60 code=406d1388 flags=0
5873.915:002b:0034:trace:seh:call_vectored_handlers handler at 0x6f46fa60 returned ffffffff
err:   DxvkMemoryAllocator: Memory allocation failed
  Size:      24000000
  Alignment: 256
  Mem flags: 0x7
  Mem types: 0x681
err:   Heap 0: 263 MB allocated, 8 MB used, 263 MB allocated (driver), 1345 MB available (driver), 2048 MB total
err:   Heap 1: 333 MB allocated, 188 MB used, 352 MB allocated (driver), 5910 MB available (driver), 5910 MB total
err:   DxvkMemoryAllocator: Memory allocation failed
err:   DxvkMemoryAllocator: Memory allocation failed
  Size:      32000000
  Alignment: 256
  Mem flags: 0x7
  Mem types: 0x681
err:   Heap 0: 263 MB allocated, 8 MB used, 263 MB allocated (driver), 1345 MB available (driver), 2048 MB total
err:   Heap 1: 333 MB allocated, 199 MB used, 352 MB allocated (driver), 5910 MB available (driver), 5910 MB total
5874.007:002b:0034:trace:seh:MSVCRT_raise (22)
terminate called after throwing an instance of 'dxvk::DxvkError'
Unable to read VR Path Registry from C:\users\steamuser\Local Settings\Application Data\openvr\openvrpaths.vrpath
Setting breakpad minidump AppID = 212480
Steam_SetMinidumpSteamID:  Caching Steam ID:  76561198193661422 [API loaded no]
5874.010:002b:0034:trace:seh:raise_exception code=c0000005 flags=0 addr=0x7bc3d311 ip=7bc3d311 tid=0034
5874.010:002b:0034:trace:seh:raise_exception  info[0]=00000000
5874.010:002b:0034:trace:seh:raise_exception  info[1]=0000023c
5874.010:002b:0034:trace:seh:raise_exception  eax=01e6f3cc ebx=7bcedff4 ecx=01e6f3b0 edx=ffffffff esi=00000228 edi=b4505700
5874.010:002b:0034:trace:seh:raise_exception  ebp=01e6f398 esp=01e6f380 cs=0023 ds=002b es=002b fs=0063 gs=006b flags=00010202
5874.010:002b:0034:trace:seh:call_vectored_handlers calling handler at 0x624fff60 code=c0000005 flags=0
5874.010:002b:0034:trace:seh:call_vectored_handlers handler at 0x624fff60 returned 0
5874.010:002b:0034:trace:seh:call_stack_handlers calling handler at 0x929fe8 code=c0000005 flags=0
5874.010:002b:0034:trace:seh:call_stack_handlers handler at 0x929fe8 returned 1
5874.010:002b:0034:trace:seh:call_stack_handlers calling handler at 0x78b2b65a code=c0000005 flags=0
5874.010:002b:0034:trace:seh:call_stack_handlers handler at 0x78b2b65a returned 1
5874.010:002b:0034:trace:seh:call_stack_handlers calling handler at 0x78b2b65a code=c0000005 flags=0
5874.010:002b:0034:trace:seh:call_stack_handlers handler at 0x78b2b65a returned 1
5874.010:002b:0034:trace:seh:call_stack_handlers calling handler at 0x7bcab590 code=c0000005 flags=0
5874.010:002b:0034:trace:seh:__regs_RtlUnwind code=c0000005 flags=2
5874.010:002b:0034:trace:seh:__regs_RtlUnwind eax=00000000 ebx=7bcab400 ecx=01e6f328 edx=01e6f52c esi=01e6f52c edi=7bcab400
5874.010:002b:0034:trace:seh:__regs_RtlUnwind ebp=01e6eee8 esp=01e6eec8 eip=7bcab45d cs=0023 ds=002b fs=0063 gs=006b flags=00000202
5874.010:002b:0034:trace:seh:__regs_RtlUnwind calling handler at 0x7bc85ef0 code=c0000005 flags=2
5874.010:002b:0034:trace:seh:__regs_RtlUnwind handler at 0x7bc85ef0 returned 1
5874.010:002b:0034:trace:seh:__regs_RtlUnwind calling handler at 0x929fe8 code=c0000005 flags=2
5874.010:002b:0034:trace:seh:__regs_RtlUnwind handler at 0x929fe8 returned 1
5874.010:002b:0034:trace:seh:__regs_RtlUnwind calling handler at 0x78b2b65a code=c0000005 flags=2
5874.010:002b:0034:trace:seh:__regs_RtlUnwind handler at 0x78b2b65a returned 1
5874.010:002b:0034:trace:seh:__regs_RtlUnwind calling handler at 0x78b2b65a code=c0000005 flags=2
5874.010:002b:0034:trace:seh:__regs_RtlUnwind handler at 0x78b2b65a returned 1
5875.091:001f:0023:trace:loaddll:free_modref Unloaded module L"C:\\windows\\system32\\winehid.sys" : builtin
5875.091:001f:0023:trace:loaddll:free_modref Unloaded module L"C:\\windows\\system32\\hidclass.sys" : builtin
5875.091:001f:0023:trace:loaddll:free_modref Unloaded module L"C:\\windows\\system32\\hid.dll" : builtin
pid 32033 != 32030, skipping destruction (fork without exec?)

**Comment by vinifmor** [comment]
**0.13f** fixed the initialization problem, but the game keeps crashing when a race starts. The last proton log lines:

1468.374:002b:0034:trace:seh:__regs_RtlUnwind calling handler at 0x7bc85ef0 code=40010006 flags=2
1468.374:002b:0034:trace:seh:__regs_RtlUnwind handler at 0x7bc85ef0 returned 1
err:   DxvkMemoryAllocator: Memory allocation failed
  Size:      524288
  Alignment: 256
  Mem flags: 0xe
  Mem types: 0x681
err:   Heap 0: 461 MB allocated, 310 MB used, 475 MB allocated (driver), 1827 MB budget (driver), 2048 MB total
err:   Heap 1: 576 MB allocated, 573 MB used, 604 MB allocated (driver), 5910 MB budget (driver), 5910 MB total
1468.480:002b:0034:trace:seh:MSVCRT_raise (22)
Racing Transformed\ASN_App_PcDx9_Final.exe: ../nptl/pthread_mutex_lock.c:442: __pthread_mutex_cond_lock_full: Assertion `robust || (oldval & FUTEX_OWNER_DIED) == 0' failed.
1468.481:002b:0042:warn:seh:setup_exception_record exception outside of stack limits in thread 0042 eip f7fdbde9 esp 02e624a0 stack 0x2d62000-0x2e60000
1468.481:002b:0042:trace:seh:raise_exception code=80000101 flags=1 addr=0xf7fdbde9 ip=f7fdbde9 tid=0042
1468.481:002b:0042:trace:seh:raise_exception  eax=00000000 ebx=00000002 ecx=02e624bc edx=00000000 esi=00000008 edi=00000000
1468.481:002b:0042:trace:seh:raise_exception  ebp=02e624bc esp=02e624a0 cs=0023 ds=002b es=002b fs=0063 gs=006b flags=00000282
1468.481:002b:0042:trace:seh:call_vectored_handlers calling handler at 0x6f46fa60 code=80000101 flags=1
1468.481:002b:0042:trace:seh:call_vectored_handlers handler at 0x6f46fa60 returned 0
1468.481:002b:0042:trace:seh:call_vectored_handlers calling handler at 0x624fe990 code=80000101 flags=1
1468.481:002b:0042:trace:seh:call_vectored_handlers handler at 0x624fe990 returned 0
1468.481:002b:0042:trace:seh:call_stack_handlers calling handler at 0x7bcab4a0 code=80000101 flags=1
wine client error:42: write: Bad file descriptor
terminate called after throwing an instance of 'dxvk::DxvkError'
Unable to read VR Path Registry from C:\users\steamuser\Local Settings\Application Data\openvr\openvrpaths.vrpath
1469.613:001f:0023:trace:loaddll:free_modref Unloaded module L"C:\\windows\\system32\\winehid.sys" : builtin
1469.613:001f:0023:trace:loaddll:free_modref Unloaded module L"C:\\windows\\system32\\hidclass.sys" : builtin
1469.613:001f:0023:trace:loaddll:free_modref Unloaded module L"C:\\windows\\system32\\hid.dll" : builtin
pid 6039 != 6037, skipping destruction (fork without exec?)

**Comment by NerosTie** [comment]
None of these issues seem to exist anymore.

![Screenshot_20190925_124603](https://user-images.githubusercontent.com/4572452/65594355-dd263a00-df92-11e9-8666-6f8a274953a5.png)
![Screenshot_20190925_124701](https://user-images.githubusercontent.com/4572452/65594364-e2838480-df92-11e9-97c9-f19e15bf8577.png)

I think it can be closed.

**Comment by vinifmor** [comment]
I tried yesterday with the native Steam client and Proton 4.11, but the game did not boot. It seems d3d9.dll is not found. The steps that I followed below:
- downloaded d9vk021 release
- I typed the following commands within the extracted folder:
`
export WINEPREFIX=/home/$USER/.steam/steam/steamapps/compatdata/212480/pfx
./setup_dxvk.sh install
`

Proton logs (error):
`
931.819:0035:0036:err:module:import_dll Library d3d9.dll (which is needed by L"Z:\\home\\$USER\\.local\\share\\Steam\\steamapps\\common\\Sonic & All-Stars Racing Transformed\\ASN_App_PcDx9_Final.exe") not found
`

Have I done anything wrong ?

**Comment by NerosTie** [comment]
Why you don't use this parameter?
```PROTON_USE_D9VK=1```

**Comment by vinifmor** [comment]
Because proton would override the installed version (0.21) by its own (0.21.RC-P)

**Comment by NerosTie** [comment]
Backup the d3d9.dll inside the proton folder and make a symlink to the dll from d9vk. No need to install anything.

Anyway, the game works, I think the issue can be closed.

**Comment by vinifmor** [comment]
@NerosTie, I could use **0.21** with Proton and it works indeed, but not a 100%. I got some crashes in some levels (like "Samba de Amigo"), and also found something strange. If you remove the associated game's **shadercache folder** to d9vk rebuild it with better shaders, the characters are rendered as shadows.

**Comment by Joshua-Ashton** [comment]
I managed to play 3 races across several races including Samba de Amigo and had no errors.

**Comment by vinifmor** [comment]
@Joshua-Ashton, try to delete the game shader cache folder and play, please. Here are some pictures of what happens in my machine. This only happens to this game of all my DX9 games. 

* Select screen
![212480_20191021233849_1](https://user-images.githubusercontent.com/4312414/67256259-73e3fa80-f45c-11e9-9464-cbeaa7f14e06.png)
![212480_20191021233853_1](https://user-images.githubusercontent.com/4312414/67256262-7b0b0880-f45c-11e9-810d-e6fc6754d6d9.png)

# Loading Samba de Amigo
![212480_20191021233903_1](https://user-images.githubusercontent.com/4312414/67256277-91b15f80-f45c-11e9-88c2-f6782087b164.png)

# Samba de Amigo preview
![212480_20191021233908_1](https://user-images.githubusercontent.com/4312414/67256291-9ece4e80-f45c-11e9-9061-1de1fba6ec7d.png)
![212480_20191021233918_1](https://user-images.githubusercontent.com/4312414/67256295-a42b9900-f45c-11e9-981f-a834f6bc8067.png)

# Main screen after exiting Samba de Amigo
![212480_20191021233933_1](https://user-images.githubusercontent.com/4312414/67256303-b0aff180-f45c-11e9-8e95-bac242a21cdf.png)

My system spec:
OS: Manjaro LinuxKERNEL: 5.3.6-1-MANJAROCPU: Intel Core i5-4690S @ 3.20GHzGPU: NVIDIA GeForce GTX 970GPU DRIVER: NVIDIA 435.21RAM: 16 GB

Thank you for your great effort in this project. It's really helping Linux gaming. Cheers.

**Comment by Joshua-Ashton** [comment]
Don't use Wines d3dcompiler... stuff like this will happen. This is not my bug.

**Comment by vinifmor** [comment]
I'm not using. The launch options is set as **PROTON_USE_D9VK=1 %command%**.

**Comment by Joshua-Ashton** [comment]
D9VK does not replace d3dcompiler.

**Comment by vinifmor** [comment]
So it is probably automatic set during the game installation process. Is it possible to uninstall it from the prefix ?

**Comment by NerosTie** [comment]
Try to install d3dcompiler with protontricks.

**Comment by Joshua-Ashton** [comment]
Does the same issue happen on Windows with D9VK and/or Native?

**Comment by vinifmor** [comment]
I'm currently with no Window to test. But I reinstalled the game yesterday using D9VK from the beginning and the graphics issues were gone. But configuring the game to run with the maximum settings makes it crashes when a level is loaded. I have tried with Samba de Amigo ( again ), and the game hangs in a black screen. Here are the proton logs:

err:   DxvkMemoryAllocator: Memory allocation failed
err:     Size:      2097152
err:     Alignment: 256
err:     Mem flags: 0xe
err:     Mem types: 0x681
err:   Heap 0: 456 MB allocated, 304 MB used, 471 MB allocated (driver), 3787 MB budget (driver), 4096 MB total
err:   Heap 1: 643 MB allocated, 642 MB used, 697 MB allocated (driver), 11920 MB budget (driver), 11920 MB total
1420.753:0032:0038:trace:seh:MSVCRT_raise (22)
terminate called after throwing an instance of 'dxvk::DxvkError'
Unable to read VR Path Registry from C:\users\steamuser\Local Settings\Application Data\openvr\openvrpaths.vrpath
Setting breakpad minidump AppID = 212480
Steam_SetMinidumpSteamID:  Caching Steam ID:  76561198193661422 [API loaded no]`
`
